triton-vm/src/stark.rs

```diff
@@ -602,210 +762,127 @@ impl Stark {
         prof_start!(maybe_profiler, "Merkle verify (extension tree)");
         if !MerkleTree::<StarkHasher, Maker>::verify_authentication_structure_from_leaves(
             extension_tree_merkle_root,
-            &revealed_current_and_next_row_indices,
+            &revealed_current_row_indices,
             &leaf_digests_ext,
             &auth_paths_ext,
         ) {
             bail!("Failed to verify authentication path for extension codeword");
         }
         prof_stop!(maybe_profiler, "Merkle verify (extension tree)");
 
-        // Verify Merkle authentication path for combination elements
-        prof_start!(maybe_profiler, "Merkle verify (combination tree)");
-        let revealed_combination_leafs =
+        prof_start!(maybe_profiler, "Merkle verify (combined quotient)");
+        let revealed_quotient_elements =
             proof_stream.dequeue()?.as_revealed_combination_elements()?;
-        let revealed_combination_digests = revealed_combination_leafs
+        let revealed_quotient_digests = revealed_quotient_elements
             .par_iter()
-            .map(StarkHasher::hash)
+            .map(|&x| x.into())
             .collect::<Vec<_>>();
```

Comment:
I'm not sure what these three lines are supposed to do.
```
let revealed_quotient_digests = revealed_quotient_elements
            .par_iter()
            .map(|&x| x.into())
```
Maybe add an explicit type to clarify?

Reply:
I have a added the following comment instead:
```
// Interpret the leaves, which are XFieldElements, as Digests, without hashing.
```

Does that clarify the operation sufficiently?